src/validation.cpp

```diff
@@ -340,21 +340,29 @@ void CChainState::MaybeUpdateMempoolForReorg(
     // the disconnectpool that were added back and cleans up the mempool state.
     m_mempool->UpdateTransactionsFromBlock(vHashUpdate);
 
-    const auto check_final_and_mature = [this, flags=STANDARD_LOCKTIME_VERIFY_FLAGS](CTxMemPool::txiter it)
+    // Check whether the transaction is still final and, if it spends a coinbase output, mature.
+    // Update the entry's cached LockPoints if needed. If this returns true, the tx would be invalid
+    // in the next block, so the mempool must remove this entry and all of its descendants.
+    const auto filter_final_and_mature = [this, flags=STANDARD_LOCKTIME_VERIFY_FLAGS](CTxMemPool::txiter it)
```

Comment:
nit: this sounds like we're getting rid of final/mature txs.

a little longer but `filter_for_final_and_mature` makes it less ambiguous to me

Reply:
I've updated the comments to make it more clear that this is meant to be a predicate for filtering, so returning `true` for something we want to filter out is natural.